cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/WindowAggregationDefinition.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Specifies how a window aggregation operation should be executed.
+ */
+public class WindowAggregationDefinition {
+  private List<Expression> partitionExpressions;
+  private Map<String, Expression> selectExpressions;
+  private List<Expression> orderByExpressions;
+  private String windowFrameType;
```

Comment:
I think this should be an enumeration. There is a finite number of window frame types we support, thus we can make this an enum which outlines every supported window frame type

Reply:
Changed the window frame type to enum